### PR TITLE
Use `repr` to temporary fix `test_subprocess.py::test_print`

### LINKED ISSUE #800: [UT] regression in test_subprocess.py with the PTDB 0.5.3
12 tests cases are failing:

```
2024-04-02T21:44:57.4181029Z =========================== short test summary info ============================
2024-04-02T21:44:57.4181419Z FAILED language/test_subprocess.py::test_print[device_print-int16] - assert False
2024-04-02T21:44:57.4182039Z FAILED language/test_subprocess.py::test_print[device_print-long] - assert False
2024-04-02T21:44:57.4182626Z FAILED language/test_subprocess.py::test_print[print-int32] - assert False
2024-04-02T21:44:57.4183238Z FAILED language/test_subprocess.py::test_print[device_print-float32] - assert False
2024-04-02T21:44:57.4183857Z FAILED language/test_subprocess.py::test_print[device_print-int8] - assert False
2024-04-02T21:44:57.4184439Z FAILED language/test_subprocess.py::test_print[device_print-int32] - assert False
2024-04-02T21:44:57.4185023Z FAILED language/test_subprocess.py::test_print[device_print-float16] - assert False
2024-04-02T21:44:57.4185590Z FAILED language/test_subprocess.py::test_print[device_print-float64] - assert False
2024-04-02T21:44:57.4186209Z FAILED language/test_subprocess.py::test_print[device_print-uint8] - assert False
2024-04-02T21:44:57.4186820Z FAILED language/test_subprocess.py::test_print[device_print_hex-int16] - assert False
2024-04-02T21:44:57.4187416Z FAILED language/test_subprocess.py::test_print[device_print_hex-int32] - assert False
2024-04-02T21:44:57.4188022Z FAILED language/test_subprocess.py::test_print[device_print_hex-int64] - assert False
2024-04-02T21:44:57.4188461Z ======================== 12 failed, 21 passed in 24.07s ========================
```

### PR DESCRIPTION
**Pull request by anmyachev**
Closes #800

I accidentally discovered this effect while I was working on https://github.com/intel/intel-xpu-backend-for-triton/issues/1082. It seems that accessing the tensor elements via `repr` builtin python function stimulates the execution of the kernel and obtaining output from it.

Although the behavior is different from other backends, it allows us to at least somehow test this feature right now.

### CODE REVIEW

File: python/test/unit/language/test_subprocess.py [hunk 1]
```diff
@@ -101,7 +106,7 @@ def test_print(func_type: str, data_type: str):
         if delta == 0:
             continue
         print(f'Expected line "{line}" {expected_lines[line]} time(s), but saw {actual_lines[line]} time(s)')
-    assert all(delta == 0 for delta in diff.values())
+    assert all(delta == 0 for delta in diff.values()), f"{actual_lines=}\n\n\n{expected_lines=}"
```
Comment (anmyachev) [inline python/test/unit/language/test_subprocess.py:109]:
```suggestion
    assert all(delta == 0 for delta in diff.values())
```

File: python/test/unit/language/test_subprocess.py [hunk 2]
```diff
@@ -38,6 +41,8 @@ def is_interpreter():
     ("device_print_pointer", "int32"),
 ])
 def test_print(func_type: str, data_type: str):
+    if data_type == "float64" and not tr.driver.active.get_current_target().arch['has_fp64']:
```
Comment (anmyachev) [inline python/test/unit/language/test_subprocess.py:44]: For A770

Comment (YarShev) [inline python/test/unit/language/test_subprocess.py:44]:
```suggestion
    if device == "xpu" and data_type == "float64" and not tr.driver.active.get_current_target().arch['has_fp64']:
```

Comment (anmyachev) [reply python/test/unit/language/test_subprocess.py:44]: This test is run only for xpu (according to how tensors are constructed in `print_helper.py` file) so for now it is not necessary.

Comment (YarShev) [reply python/test/unit/language/test_subprocess.py:44]: What if we change the way of constructing tensors in `print_helper.py`? I think we don't want to waste time for digging into the test why it fails in future. Anyway, let others weigh this in.

Comment (whitneywhtsang) [reply python/test/unit/language/test_subprocess.py:44]: I also prefer @YarShev's suggestion, as it becomes obvious that this check is added by us, and more future proof.

File: python/test/unit/language/print_helper.py [hunk 3]
```diff
@@ -116,6 +116,10 @@ def test_print(func: str, data_type: str):
     else:
         assert f"Unknown kernel: {func}"
 
+    # FIXME: remove trigger to get output from kernel for xpu
```
Comment (whitneywhtsang) [inline python/test/unit/language/print_helper.py:119]: According to the original investigation, the problem can also be workaround by waiting for the event explicitly, should we do that instead? @quintinwang5

Comment (quintinwang5) [reply python/test/unit/language/print_helper.py:119]: But we cannot wait only for print kernels. Maybe it's not a good idea.

Comment (whitneywhtsang) [reply python/test/unit/language/print_helper.py:119]: What's your opinion on the `repr` workaround suggested in this PR?

Comment (quintinwang5) [reply python/test/unit/language/print_helper.py:119]: I think it's good for UT. But `tl.print` in real kernels may not work still.

File: python/test/unit/language/test_subprocess.py [hunk 4]
```diff
@@ -37,8 +40,11 @@ def is_interpreter():
     ("device_print_hex", "int64"),
     ("device_print_pointer", "int32"),
 ])
-def test_print(func_type: str, data_type: str):
-    proc = subprocess.Popen([sys.executable, print_path, func_type, data_type], stdout=subprocess.PIPE,
+@pytest.mark.parametrize("device", ['xpu'])
```
Comment (whitneywhtsang) [inline python/test/unit/language/test_subprocess.py:43]: Does it work with this line removed?
As device can be given by https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/python/test/unit/conftest.py.

Comment (anmyachev) [reply python/test/unit/language/test_subprocess.py]: yes

**Comment by anmyachev** [comment]
@whitneywhtsang should i do here anything else?

File: python/test/unit/language/print_helper.py
Comment (whitneywhtsang) [inline python/test/unit/language/print_helper.py:1]: Can non-xpu specific changes in this file be upstreamed?

File: python/test/unit/language/print_helper.py [hunk 5]
```diff
@@ -116,11 +116,16 @@ def test_print(func: str, data_type: str):
     else:
         assert f"Unknown kernel: {func}"
 
+    if device == "xpu":
+        # FIXME: remove trigger to get output from kernel
+        repr(x)
+        repr(y)
```
Comment (whitneywhtsang) [inline python/test/unit/language/print_helper.py:122]: Please add a comment in https://github.com/intel/intel-xpu-backend-for-triton/issues/800 to remove this once the original problem is resolved.

**Review by whitneywhtsang — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

File: python/test/unit/language/print_helper.py (hunk 5 above)
Comment (anmyachev) [reply python/test/unit/language/print_helper.py:122]: https://github.com/intel/intel-xpu-backend-for-triton/issues/800#issuecomment-2230475557

File: python/test/unit/language/print_helper.py
Comment (anmyachev) [reply python/test/unit/language/print_helper.py:1]: https://github.com/triton-lang/triton/pull/4333

**Pull request merged**